ethpm_types/sourcemap.py

```diff
@@ -0,0 +1,162 @@
+from typing import Dict, Iterator, List, Optional, Union
+
+from ethpm_types.base import BaseModel
+
+
+class SourceMapItem(BaseModel):
+    """
+    An object modeling a node in a source map; useful for mapping
+    the source map string back to source code.
+    """
+
+    # NOTE: `None` entry means this path was inserted by the compiler during codegen
+    start: Optional[int]
+    """
+    The byte-offset start of the range in the source file.
+    """
+
+    length: Optional[int]
+    """
+    The byte-offset length.
+    """
+
+    contract_id: Optional[int]
+    """
+    The source identifier.
+    """
+
+    jump_code: str
+    """
+    An identifier for whether a jump goes into a function, returns from a function,
+    or is part of a loop.
+    """
+    # NOTE: ignore "modifier_depth" keyword introduced in solidity >0.6.x
+
+    @classmethod
+    def parse_str(cls, src_str: str, previous: Optional["SourceMapItem"] = None) -> "SourceMapItem":
```

Comment:
I don't want this to get lost in the diff -

refactored somewhat to allow parsing individual source map items because sometimes they occur in other objects, such as the AST JSON